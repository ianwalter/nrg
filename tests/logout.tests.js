const { test } = require('@ianwalter/bff')
const app = require('../examples/accounts')
const { accounts, password } = require('../examples/accounts/seeds/01_accounts')

const [julian] = accounts

test('Logout when not logged in', async ({ expect }) => {
  const response = await app.test('/logout').delete()
  expect(response.status).toBe(200)
  // TODO: enable CSRF logic in NOD_ENV=test
  // expect(response.body.csrfToken).toBeDefined()
})

test('Logout when logged in', async ({ expect }) => {
  // Login.
  let response = await app.test('/login').post({ ...julian, password })

  // Verify account data can be retrieved.
  response = await app.test('/account', response).get()
  expect(response.status).toBe(200)
  expect(response.body).toMatchSnapshot()

  // Logout.
  response = await app.test('/logout', response).delete()
  expect(response.status).toBe(200)
  // expect(response.body.csrfToken).toBeDefined()

  // Verify account data cannot be retrieved.
  response = await app.test('/account', response).get()
  expect(response.status).toBe(401)
})
