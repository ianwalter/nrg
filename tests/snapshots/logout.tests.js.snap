// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout when logged in 1`] = `
Object {
  "email": "general_user_test@example.com",
  "emailVerified": true,
  "firstName": "General User",
  "id": 1,
  "lastName": "Test",
}
`;
